cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/CloudSqlPostgreSqlClient.java

```diff
@@ -0,0 +1,174 @@
+       database,instanceConnectionName, username, password);
+    Connection conn = DriverManager.getConnection(jdbcUrl);
+    System.out.println("Connected to the database successfully");
+    return conn;
+  }
+
+  public static int countRecord(String table, String schema) throws SQLException, ClassNotFoundException {
+    String countQuery = "SELECT COUNT(*) as total FROM " + schema + "." + table;
+    try (Connection connect = getCloudSqlConnection();
+         Statement statement = connect.createStatement();
+         ResultSet rs = statement.executeQuery(countQuery)) {
+      int num = 0;
+      while (rs.next()) {
+        num = (rs.getInt(1));
+      }
+      return num;
+    }
+  }
+
+  /**
+   * Extracts entire data from source and target tables.
+   * @param sourceTable table at the source side
+   * @param targetTable table at the sink side
+   * @return true if the values in source and target side are equal
+   */
+  public static boolean validateRecordValues(String sourceTable, String targetTable, String schema)
+    throws SQLException, ClassNotFoundException {
+    String getSourceQuery = "SELECT * FROM " + schema + "." + sourceTable;
+    String getTargetQuery = "SELECT * FROM " + schema + "." + targetTable;
+    try (Connection connect = getCloudSqlConnection()) {
+      connect.setHoldability(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      Statement statement1 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                                                     ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      Statement statement2 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                                                     ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      ResultSet rsSource = statement1.executeQuery(getSourceQuery);
+      ResultSet rsTarget = statement2.executeQuery(getTargetQuery);
+      return compareResultSetData(rsSource, rsTarget);
+    }
+  }
+
+  /**
+   * Compares the result Set data in source table and sink table..
+   * @param rsSource result set of the source table data
+   * @param rsTarget result set of the target table data
+   * @return true if rsSource matches rsTarget
+   */
+  public static boolean compareResultSetData(ResultSet rsSource, ResultSet rsTarget) throws SQLException {
+    ResultSetMetaData mdSource = rsSource.getMetaData();
+    ResultSetMetaData mdTarget = rsTarget.getMetaData();
+    int columnCountSource = mdSource.getColumnCount();
+    int columnCountTarget = mdTarget.getColumnCount();
+    Assert.assertEquals("Number of columns in source and target are not equal",
+                        columnCountSource, columnCountTarget);
+    while (rsSource.next() && rsTarget.next()) {
+      int currentColumnCount = 1;
+      while (currentColumnCount <= columnCountSource) {
+        String columnTypeName = mdSource.getColumnTypeName(currentColumnCount);
```

Comment:
Is this logic of comparing result set copied from PostgreSQL ? How do we even know we just have to cover timestamp in a separate case and rest we can compare as a string.